src/trace/hit_cylinder.c

```diff
@@ -0,0 +1,55 @@
+#include <math.h>
+#include "trace.h"
+#include "vector3.h"
+
+static t_bool	check_cylinder(
+	t_obj_list objects[], t_ray *ray, t_hit_record *rec, double root)
+{
+	t_cylinder	*cy;
+	double		p_height;
+
+	cy = (t_cylinder *)objects->object;
+	if (isnan(root) || root < rec->tmin || rec->tmax < root)
+		return (FALSE);
+	rec->t = root;
+	rec->p = ray_at(ray, root);
+	p_height = vec3_dot(vec3_minus(rec->p, cy->center), cy->normal);
+	if (p_height < 0 || p_height > cy->height)
+		return (FALSE);
+	rec->normal = vec3_unit(vec3_minus(rec->p, vec3_plus(cy->center,
+					vec3_mult_scalar(cy->normal, vec3_dot(
+							vec3_minus(rec->p, cy->center), cy->normal)))));
+	rec->color = objects->color;
+	set_face_normal(ray, rec);
```

Comment:
face_normal_set로 바꾸시면 될 듯??
```suggestion
	face_normal_set(ray, rec);
```